source/agora/common/MerkleTree.d

```diff
@@ -0,0 +1,233 @@
+/*******************************************************************************
+
+    Contains MerkleTree that makes binary tree of transactions.
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.node.MerkleTree;
+
+import agora.common.Data;
+import agora.common.Hash;
+import agora.common.Transaction;
+
+import std.algorithm;
+
+
+/// Ditto
+public struct MerkleTree
```

Comment:
This seems to be redundant with `MerkleNode` now, isn't it ?

Reply:
@Geod24 I added several method